tests/storage/export.go

```diff
@@ -0,0 +1,541 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func() bool {
+			export, err = virtClient.VirtualMachineExport(pvc.Namespace).Get(context.Background(), export.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			condReady := false
+			if export.Status != nil {
+				for _, cond := range export.Status.Conditions {
+					if cond.Type == exportv1.ConditionReady && cond.Status == k8sv1.ConditionTrue {
+						condReady = true
+					}
+				}
+			}
+			return condReady
+		}, 30*time.Second, 1*time.Second).Should(BeTrue(), "export is expected to become ready")
+
+		By("Creating download pod, so we can download image")
+		targetPvc := &k8sv1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("target-pvc-%s", rand.String(12)),
+				Namespace: pvc.Namespace,
+			},
+			Spec: k8sv1.PersistentVolumeClaimSpec{
+				AccessModes:      pvc.Spec.AccessModes,
+				StorageClassName: pvc.Spec.StorageClassName,
+				Resources:        pvc.Spec.Resources,
+			},
+		}
+		By("Creating target PVC, so we can inspect if the export worked")
+		targetPvc, err = virtClient.CoreV1().PersistentVolumeClaims(targetPvc.Namespace).Create(context.Background(), targetPvc, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		caConfigMap := createCaConfigMap("export-cacerts", targetPvc.Namespace, export.Status.Links.Internal.Cert)
+		downloadPod := createDownloadPodForPvc(targetPvc, caConfigMap)
+		downloadUrl := ""
+		fileName := ""
+		for _, volume := range export.Status.Links.Internal.Volumes {
+			if volume.Name == pvc.Name {
+				for _, format := range volume.Formats {
+					if format.Format == expectedFormat {
+						downloadUrl = fmt.Sprintf(urlTemplate, format.Url, pvc.Name)
+						fileName = filepath.Base(format.Url)
+					}
+				}
+			}
+		}
+		Expect(downloadUrl).ToNot(BeEmpty())
+		Expect(fileName).ToNot(BeEmpty())
+		command := []string{
+			"curl",
+			"-L",
+			"--cacert",
+			filepath.Join(caCertPath, caBundleKey),
+			downloadUrl,
+			"--output",
+			filepath.Join(dataPath, fileName),
+		}
+		By(fmt.Sprintf("Downloading from URL: %s", downloadUrl))
+		out, stderr, err := tests.ExecuteCommandOnPodV2(virtClient, downloadPod, downloadPod.Spec.Containers[0].Name, command)
+		Expect(err).ToNot(HaveOccurred(), out, stderr)
+
+		verifyFunction(fileName, comparison, downloadPod)
+	},
+		Entry("with RAW kubevirt content type", populateKubeVirtContent, verifyKubeVirtRawContent, exportv1.KubeVirtRaw, kubevirtcontentUrlTemplate),
+		Entry("with RAW gzipped kubevirt content type", populateKubeVirtContent, verifyKubeVirtGzContent, exportv1.KubeVirtGz, kubevirtcontentUrlTemplate),
+		Entry("with archive content type", populateArchiveContent, verifyKubeVirtRawContent, exportv1.Dir, archiveDircontentUrlTemplate),
+		Entry("with archive tarred gzipped content type", populateArchiveContent, verifyArchiveGzContent, exportv1.ArchiveGz, kubevirtcontentUrlTemplate),
```

Comment:
Should we test block volume?

Reply:
Don't think we support block volumes do we? At least not in this version.

Reply:
Maybe we do. Let me see if I can add a test for it.